internal/provider/utils/utils.go

```diff
@@ -12,3 +16,14 @@ func NamespacedName(obj client.Object) types.NamespacedName {
 		Name:      obj.GetName(),
 	}
 }
+
+// Returns a partially hashed name for the string if it is more than 60 chars. Otherwise returns the original string
+func GetHashedName(name string) string {
+	if len(name) > 60 {
```

Comment:
might make sense to make this hashing consistent for all strings and not just the ones above 60 char

Reply:
Ok, everything is now hashed. It now takes up to 32 characters (arbitrary number I chose that seemed reasonable) and then adds the 8 character hash. I can increase/change how much of the original name we keep if you think we should increase it or decrease it further.

Reply:
thanks, the more chars the better imho, I think we should be able to do 48 chars without breaking anything (48 + 8, 5 for envoy, 2 for the `-`)

Reply:
Sounds good. 48 it is now which puts us right at the 63 character mark